### PR TITLE
[helm] bind release cadence of helm charts with that of the binary

### LINKED ISSUE #1392: Decouple helm chart releases from application release stream
**What happened**:
Repository is constantly spamming about new version updates of helm chart and not the application.

**What you expected to happen**:
I expect to have a notification when a new version of the **application** comes out and not of some opinionated contrib part in form of a helm chart.

**How to reproduce it (as minimally and precisely as possible)**:
Just go to https://github.com/kubernetes/kube-state-metrics/releases

**Anything else we need to know?**:

I am working on automatic updates of applications based on github releases. This is something I would like to have in kube-prometheus project and as of today I have 99% of this complete, apart of KSM piece where constant releases break everything up.

**Environment**:
- kube-state-metrics version: 
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Other info:

### PR DESCRIPTION
**Pull request by tariq1890**
**What this PR does / why we need it**:

This PR will render the helm-chart as a first-class entity of the `kube-state-metrics` binding it to the release cadence of the code base.

If this PR is merged, one can only release new Helm Chart artifacts from a release branch. This means that artifacts are created for both the binary as well as the helm chart during a release-cut.

**NOTE**: This involves a breaking change as the helm chart is renamed and the versioning is set to match that of the main project.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1392

### CODE REVIEW

File: charts/kube-state-metrics-v2/Chart.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+name: kube-state-metrics-v2
 description: Install kube-state-metrics to generate and expose cluster-level metrics
 keywords:
 - metric
 - monitoring
 - prometheus
 - kubernetes
-version: 2.13.0
-appVersion: 1.9.8
+version: 2.0.0
```
Comment (mrueg) [inline charts/kube-state-metrics-v2/Chart.yaml:9]: We should have a target in the Makefile that non-helm users can execute to update these versions before release.

File: .github/workflows/helm-cd.yaml [hunk 2]
```diff
@@ -3,7 +3,10 @@ name: Release Charts
 on:
   push:
     branches:
-      - master
+    - release*
+    tags:
+    - v1.*
```
Comment (mrueg) [inline .github/workflows/helm-cd.yaml:8]: Are you planning to include future v1 releases or should we focus on v2+ for this?

**Comment by mrueg** [comment]
Thanks for proposing the change @tariq1890 !

We should look into providing documentation around migration paths and automation for version bumps as well.

**Comment by tariq1890** [comment]
Thanks for the review @mrueg 

Closing this PR in favour of #1438. I don't seem to be able to force push to this branch anymore

**Pull request closed without merging**